torch/_dynamo/utils.py

```diff
@@ -1101,6 +1101,8 @@ def fullname(o):
             return module + "." + klass.__qualname__
 
         return fullname(x)
+    elif x == collections.namedtuple:
+        return "G['collections'].namedtuple"
```

Comment:
That's right. I have completely rewritten the way we handle guards for dict keys, as the previous way was eager (not great) and buggy. The proposed way is not lazy yet, because we instantiate most of them inside `_HashableTracker`, but I think we could really make them proper lazy.